xarray/plot/plot.py

```diff
@@ -391,19 +393,25 @@ def hist(
     yticks=None,
     xlim=None,
     ylim=None,
+    bins=None,
+    weights=None,
+    density=False,
     **kwargs,
 ):
     """
     Histogram of DataArray.
 
     Wraps :py:func:`matplotlib:matplotlib.pyplot.hist`.
 
-    Plots *N*-dimensional arrays by first flattening the array.
+    Plots *N*-dimensional arrays by first flattening the array and calculating
+    the histogram via :py:func:`DataArray.hist`.
 
     Parameters
     ----------
     darray : DataArray
-        Can have any number of dimensions.
+        Can have any number of dimensions, but will be reduced to 1 dimension
```

Comment:
Maybe you have plans for this already but will this handle 2d-histograms like the mean oxygen example in https://xhistogram.readthedocs.io/en/latest/tutorial.html#Averaging-a-variable?

Reply:
Yes, it will be able to calculate 2D (or N-D in general) histograms in `xarray.hist()`, it just won't be able to plot them in `xarray.plot.hist()`, because you need something like an `xarray.plot.imshow()` to plot the results of 2D histogram instead.

Reply:
I'm not sure whether you feel it is at all confusing to use the same word "hist" to refer both to the function that calculates the histogram and to the plotting function?

Reply:
I think it's fine. `da.hist()` vs. `da.plot.hist()` feels distinct enough for me. 
I think I prefer the functions being called the proper name, `da.histogram()` and `da.plot.histogram()`, instead of the shorthand though.

Reply:
> I think it's fine. da.hist() vs. da.plot.hist() feels distinct enough for me.

Cool.

> I think I prefer the functions being called the proper name

That would also be marginally easier for the current users of xhistogram, who have been using `xhistogram.histogram` so would only need to change the import.

However the two arguments I can see against that are:
1) Deprecation cycle - I would like consistency between the calculation function and the plot function, but the plot function already exists and is so named because it wraps `matplotlib.pyplot.hist`. Changing that would require a deprecation cycle, but I think with the changes I'm proposing here we could get away without a deprecation cycle (because we're only adding optional arguments, not changing them).
2) We have many shorthand names for high-level functions already - `cov`, `corr`, `concat`, it would not be unusually terse.

I don't really have a strong opinion though.

Reply:
The deprecation cycle wouldn't hinder your progress since I'm imagining simply just aliasing the method with a warning for a while.

numpy and dask also uses `.histogram()` for their histograms. matplotlib and pandas uses `.hist()` for the plotting, holoviews uses `.Histogram()`.
Maybe it's just easier to follow the upstream dependencies decisions then to avoid messing with the muscle memory (even though it's in my opinion "bad" muscle memory)?

And I'm not a fan of those shorthand names either, `cov`id, `corr`osion, `interp`ret? But I'm fine with just following what upstream are doing, there's something satisfying when you can just jump between the different data structures and everything just works.

Reply:
You make very good points! I think I agree with you now.

> `cov`id, `corr`osion, `interp`ret

:rofl: